src/abstractdataframe/abstractdataframe.jl

```diff
@@ -317,43 +317,14 @@ Get a data frame with the `n` last rows of `df`.
 Base.last(df::AbstractDataFrame, n::Integer) = df[max(1,nrow(df)-n+1):nrow(df), :]
 
 # get the structure of a df
-"""
-Show the structure of an AbstractDataFrame, in a tree-like format
-
-```julia
-dump(df::AbstractDataFrame, n::Int = 5)
-dump(io::IO, df::AbstractDataFrame, n::Int = 5)
-```
-
-**Arguments**
-
-* `df` : the AbstractDataFrame
-* `n` : the number of levels to show
-* `io` : optional output descriptor
-
-**Result**
-
-* nothing
-
-**Examples**
-
-```julia
-df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
-dump(df)
-```
-
-"""
-function Base.dump(io::IO, df::AbstractDataFrame, n::Int, indent)
+function Base.dump(io::IOContext, df::AbstractDataFrame, n::Int, indent)
```

Comment:
I think that's too restrictive, any `IO` object should be allowed. Anyway, can we handle this in a separate PR?

Reply:
Any `IO` is allowed via `Base` via `dump(io::IO, @nospecialize(x); maxdepth=DUMP_DEFAULT_MAXDEPTH)` (show.jl line 1717).
This is a low-level function that gets called there. As I have showed in the error message - we really never handled `IO` as trying to use it threw an error. With this change `dump(stdout, df)` should work just fine.